### PR TITLE
Added type for queryParams in fetchK8s

### PR DESCRIPTION
**Pull request by gnehapk**

### CODE REVIEW

File: frontend/public/graphql/client.ts [hunk 1]
```diff
@@ -50,5 +50,5 @@ export const fetchK8s = <R extends K8sResourceCommon = K8sResourceCommon>(
   kind: K8sKind,
   name?: string,
   ns?: string,
-  opts?: { [k: string]: string },
+  opts?: { [k: string]: string } | { queryParams: { [k: string]: string } },
```
Comment (rawagner) [inline frontend/public/graphql/client.ts:53]: lets make it even more specifc
```suggestion
  opts?: { 
    path?: string;
    queryParams?: {
      [k: string]: string;
    };
  },
```

Comment (gnehapk) [reply frontend/public/graphql/client.ts:53]: ack

Comment (gnehapk) [reply frontend/public/graphql/client.ts:53]: Shouldn't we need to add `[k: string]: string` too here for handling other options like `ns?: string, name?: string`

Comment (rawagner) [reply frontend/public/graphql/client.ts:53]: I dont like duplication.

what about 

```
export const fetchK8s = <R extends K8sResourceCommon = K8sResourceCommon>(
  kind: K8sKind,
  name?: string,
  ns?: string,
  path?: string,
  queryParams: { [key: string]: string },
```

would that work for you ?

Comment (gnehapk) [reply frontend/public/graphql/client.ts:53]: Yes, this will work for all the options that are acceptable inside `getK8sResourcePath`.

**Comment by gnehapk** [comment]
Updated the PR. @rawagner Please review

**Pull request merged**